pointerToSlice/pointerToSlice.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package pointerToSlice
+
+import (
+	"go/ast"
+	"strings"
+
+	"golang.org/x/tools/go/analysis"
+)
+
+var Analyzer = &analysis.Analyzer{
+	Name: "pointerToSlice",
+	Doc:  "check for usage of pointer to slice in function definitions",
+	Run:  run,
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	checkNode := func(expr ast.Expr) {
+		sexpr, ok := expr.(*ast.StarExpr)
```

Comment:
Can you clarify a bit why we are only looking for star expressions?

Reply:
I thought that's how you catch pointers to some type in function declarations.

Reply:
Interesting.. I thought star expressions were like `db.Exec` things, where you have an `x.y` type of thing. Anyways, if you have tested and it works, then all good with me.

Slightly out of scope: I know this repo historically hasn't had tests. But I have asked for tests in recent PRs. Would be good to have a couple of them here too, but since this is such a small PR, may not be worth the time. Up to you.

Reply:
> Interesting.. I thought star expressions were like db.Exec things, where you have an x.y type of thing. 

My understanding is that it catches anything involving a `*` but in the case of function signatures that can only mean a pointer I guess.

> Slightly out of scope: I know this repo historically hasn't had tests. But I have asked for tests in recent PRs. Would be good to have a couple of them here too, but since this is such a small PR, may not be worth the time. Up to you.

Sure, I can give it a try. Yes obviously I've tested a bit locally before submitting and everything looked okay. Of course this doesn't cover all cases of pointer to slices, just the most common (either direct `*[]SomeType` usage or `*SomeType` where `SomeType` is a slice).